pkg/models/tenant/tenant.go

```diff
@@ -118,49 +118,41 @@ func (t *tenantOperator) ListWorkspaces(user user.Info, queryParam *query.Query)
 	}
 
 	decision, _, err := t.authorizer.Authorize(listWS)
-
 	if err != nil {
 		klog.Error(err)
 		return nil, err
 	}
 
 	if decision == authorizer.DecisionAllow {
-
 		result, err := t.resourceGetter.List(tenantv1alpha2.ResourcePluralWorkspaceTemplate, "", queryParam)
-
 		if err != nil {
 			klog.Error(err)
 			return nil, err
 		}
-
 		return result, nil
 	}
 
 	workspaceRoleBindings, err := t.am.ListWorkspaceRoleBindings(user.GetName(), "")
-
 	if err != nil {
 		klog.Error(err)
 		return nil, err
 	}
 
 	workspaces := make([]runtime.Object, 0)
-
 	for _, roleBinding := range workspaceRoleBindings {
-
 		workspaceName := roleBinding.Labels[tenantv1alpha1.WorkspaceLabel]
-		workspace, err := t.resourceGetter.Get(tenantv1alpha2.ResourcePluralWorkspaceTemplate, "", workspaceName)
-
+		obj, err := t.resourceGetter.Get(tenantv1alpha2.ResourcePluralWorkspaceTemplate, "", workspaceName)
```

Comment:
ResourceGetter is big, should just use the specific resource you need, create that big object will consume much more resource